fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBMetaDataStore.java

```diff
@@ -520,10 +520,31 @@ public RecordMetaData getRecordMetaData() {
         return context.asyncToSync(FDBStoreTimer.Waits.WAIT_LOAD_META_DATA, getRecordMetaDataAsync(true));
     }
 
+    /**
+     * Save the record meta-data into the meta-data store.
+     *
+     * <p> This method will automatically add a union if the file descriptor is missing one. </p>
+     *
+     * @param fileDescriptor the file descriptor of the record meta-data
+     */
+    public void saveRecordMetaData(@Nonnull Descriptors.FileDescriptor fileDescriptor) {
```

Comment:
IMO, this is too much detail for users. Also, I'm not sure why would a user want to save a proto without a union? Do you have a usecase in mind for that? Allowing that would require a lot more plumbing In `RecordMetaDataBuilder`.

Reply:
I'm somewhat sympathetic to the idea that adding the flag would enworsen rather than embetter the clarity of the API, though it's more restrictive. (But maybe restrictive is good?)

In my head, if someone set `addUnionIfMissing` to `false`, then it would throw an error kvetching about the lack of union rather than saving it without the union. The idea there is that the user could set it to `false` (or maybe it's `false` by default?) if they wanted to avoid accidentally adding an auto-generated union type.

Reply:
yeah, if we introduce the flag, probably it should only return an error when the union is missing. My vote is lets not add it, but have no strong feelings. Let me know if you still want it.

Reply:
Hm, well, I'm a little uneasy that a naïve user might accidentally call this method, and then their local file descriptor and the one stored in the meta-data won't match (especially if they can't set a local file descriptor successfully in that state). The idea behind adding the flag is now a user has to explicitly set a weird flag before the API let's them do something that could be bad (in theory). I'm not sure if that would actually stop someone in practice.

I'm not sure I'm married to that idea, though.

Reply:
Yeah, none are ideal. Then, lets just keep what we have for now. We can always change it if need be. Thanks.

Reply:
Well, I'm not sure we can change it (without breaking the API), which is why I want to try to get it right the first time. In particular, changing the default behavior from adding to not adding the auto-generated union descriptor could be problematic.

The more I think about it, the more I think I can get behind the union descriptor being included by default (for this API call). So, I suppose it is fine, then.

This method should probably be marked as `EXPERIMENTAL`, though.

Reply:
done.